### PR TITLE
dspa is not name specific anymore

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://issues.redhat.com/browse/RHOAIENG-945

## Description

- name  for new dspa is now `dspa-def`
- we now  fetch the list of dspa resources and get the first one. Empty list returns null
- adjusted get cr and delete cr to accept a name and also storing cr name in the pipelines API

## How Has This Been Tested?

create a pipelines server, make sure it uses new name
test that it grabs the dspa correctly

I understand there will possible issues where a user incorrectly has multiple dspas in their project. We will grab the first one so deletion will act funny, and if the add a new one that ends up at the top of the list, then their will be weird issues too. The frontend already prevents users from making multiple, but the backend team needs to also prevent this on their end.

## Test Impact

Make sure all old test pass and grab the dspa

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
waiting to see what tests fail if any. I am guessing we need to add a new mock

File: frontend/src/concepts/pipelines/const.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/const.ts:2]:
```suggestion
export const PIPELINE_DEFINITION_NAME = 'dspa';
```
This was my suggestion -- I don't think we need anything special here. Shorter the better -- but this is a minor nit -- since every resource will suffer the same restrictions of the 63 characters from the Route object

**Review by andrewballantyne — COMMENTED** [review]
I lost my cluster mid test -- will have to test it tomorrow. Looking good though.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
This works really well, I have one suggestion for just minor improvement to performance. Let me know if you have any questions.

Also you'll need a rebase after the last PR merge.

File: frontend/src/concepts/pipelines/context/usePipelineNamespaceCR.ts
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/context/usePipelineNamespaceCR.ts:42]: sure i will take a look

**Comment by andrewballantyne** [comment]
I think I screwed up with my merge effort @Gkrumbach07 and caused a conflict here 😞 sorry!

**Comment by Gkrumbach07** [comment]
np @andrewballantyne - rebased, should be good

**Comment by Gkrumbach07** [comment]
/hold

need to fix a issue with route naming as that is constant

**Review by jpuzz0 — APPROVED** [review]
Verified by creating a project, a pipeline server, and was able to successfully make v2 requests against that server (even though the UI isn't setup to consume these resources yet).

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**
